File: i2f-jdk/i2f-swl/src/main/java/i2f/swl/core/key/SwlKeyManager.java
package i2f.swl.core.key;

import i2f.crypto.std.encrypt.asymmetric.key.AsymKeyPair;

/**
 * @author Ice2Faith
 * @date 2024/8/4 21:42
 * @desc
 */
public interface SwlKeyManager {
    AsymKeyPair getDefaultSelfKeyPair();

    String getDefaultSelfAsymSign();

    void setDefaultSelfAsymSign(String selfAsymSign);

    AsymKeyPair getSelfKeyPair(String selfAsymSign);

    void setSelfKeyPair(String selfAsymSign, AsymKeyPair keyPair);

    String getDefaultOtherPublicKey();

    String getDefaultOtherAsymSign();

    void setDefaultOtherAsymSign(String otherAsymSign);

    String getOtherPublicKey(String otherAsymSign);

    void setOtherPublicKey(String otherAsymSign, String publicKey);
}

File: i2f-jdk/i2f-swl/web/i2f-swl/src/i2f-turbo-web/i2f-swl/core/core/key/SwlKeyManager.js
/**
 * @return {SwlKeyManager}
 * @interface {SwlKeyManager}
 */
function SwlKeyManager() {

}


/**
 * @return {AsymKeyPair}
 */
SwlKeyManager.prototype.getDefaultSelfKeyPair = function () {

}

/**
 * @return {String}
 */
SwlKeyManager.prototype.getDefaultSelfAsymSign = function () {

}

/**
 *
 * @param selfAsymSign {String}
 * @return {void}
 */
SwlKeyManager.prototype.setDefaultSelfAsymSign = function (selfAsymSign) {

}

/**
 *
 * @param selfAsymSign {String}
 * @return {AsymKeyPair}
 */
SwlKeyManager.prototype.getSelfKeyPair = function (selfAsymSign) {

}

/**
 *
 * @param selfAsymSign {String}
 * @param keyPair {AsymKeyPair}
 * @return {void}
 */
SwlKeyManager.prototype.setSelfKeyPair = function (selfAsymSign, keyPair) {

}

/**
 * @return {String}
 */
SwlKeyManager.prototype.getDefaultOtherPublicKey = function () {

}

/**
 * @return {String}
 */
SwlKeyManager.prototype.getDefaultOtherAsymSign = function () {

}

/**
 *
 * @param otherAsymSign {String}
 * @return {void}
 */
SwlKeyManager.prototype.setDefaultOtherAsymSign = function (otherAsymSign) {

}

/**
 *
 * @param otherAsymSign {String}
 * @return {String}
 */
SwlKeyManager.prototype.getOtherPublicKey = function (otherAsymSign) {

}

/**
 *
 * @param otherAsymSign {String}
 * @param publicKey {String}
 * @return {void}
 */
SwlKeyManager.prototype.setOtherPublicKey = function (otherAsymSign, publicKey) {

}

export default SwlKeyManager
